lib/internal/streams/destroy.js

```diff
@@ -55,6 +55,8 @@ function undestroy() {
     this._writableState.destroyed = false;
     this._writableState.ended = false;
     this._writableState.ending = false;
+    this._writableState.finalCalled = false;
+    this._writableState.prefinished = false;
```

Comment:
can you add a unit test for those? Maybe also place them in a separate commit, if we want to backport them separately.

Reply:
@mcollina Do you know where those tests are? In any case, these lines are tested in the sense that tests do fail without them...

Reply:
We need test that can be run as part of `readable-stream`.

Here are the current tests:

```
test/parallel/test-stream-duplex-destroy.js
test/parallel/test-stream-readable-destroy.js
test/parallel/test-stream-transform-destroy.js
test/parallel/test-stream-writable-destroy.js
```